src/functional/ffront/ast_passes/unchain_compares.py

```diff
@@ -0,0 +1,71 @@
+# GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2021, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import ast
+import copy
+
+
+class UnchainComparesPass(ast.NodeTransformer):
+    """
+    Turn comparison chain into a tree of comparisons linked by ``&`` operations.
+
+    ``a > b < c == d ...`` is turned into ``((a > b) & ((b < c) & (c == d))`` and so on.
+    The resulting AST will have only ``Comparison`` nodes with to operands and one operation.
+    That is to say with the same structure as FOAST ``Comparison`` nodes.
+
```

Comment:
```suggestion
    Note that this pass does not respect short-circut evaluation semantics.
```

Reply:
Is that not obvious enough from the dockest example? Do we need to add this comment to all the other passes as well? All of them (including this one) return semantically equivalent code. None of them return exactly the same code (which is the point).

Reply:
If it is semantically equivalent no comment is needed I fully agree, but in this case it is not semantically equivalent Python code as some conditions are evaluated multiple times. That is what I tried to express in the note, but I assume this means the phrasing is not explicit enough.

Reply:
Can you explain what gets evaluated multiple times?

Reply:
I misread it as "does not roundtrip to the same code".